test/runtime/configMatters/comm/unordered/unorderedCopyBasic.chpl

```diff
@@ -73,6 +78,35 @@ proc srcLocalAssign(param compileTimeKnown: bool, dstLocal: bool) {
   writeln("srcLocalAssign(compileTimeKnown="+compileTimeKnown+", dstLocal="+dstLocal+")");
 }
 
+proc srcLocalConstAssign(param compileTimeKnown: bool, dstLocal: bool) {
+  var dst = iniDstValue();
+  if compileTimeKnown {
```

Comment:
I think the sense of this conditional is backwards (if compileTimeKnown generates a const; if unknown, it generates a param)?

Reply:
Not sure I'm following. In this case `compileTimeKnown` refers to whether it's possible for the compiler to know if src is local at compile time (will we codegen a narrow or wide pointer for src)

The param case is in a different function (and since params have to be "local" there is no flag for `compileTimeKnown` on that routine.)

Reply:
A cleaner way of writing this test might be:

```chpl
  var dst = iniDstValue();
  on Locales[getLocaleID(curLocale=dstLocal)] {
    const src = iniSrcValue();
    if compileTimeKnown then            assign(dst, src);
                        else on here do assign(dst, src);
    printThem(dst, src);
  }
```

(where the intent is just to go through different codegen paths since we optimize when the compiler knows src is local and call chpl_comm_unordered_get, instead of chpl_comm_unordered_copy)

Reply:
Oh, I see.  I mis-guessed that "compileTimeKnown" was being used to indicate const vs. param and sloppily scanned ahead in the code for where param was named after noting that it wasn't used here (even though, later, I saw that it was tested in a separate procedure).  My bad, ignore me.